### ISSUE TITLE
OSError: [Errno 22]

### ISSUE THREAD

**Issue by DanGM96** [rhoriguchi/tv_time_export#3]
Just tried to run this and all it got was the "Recently watched" (2 out of hundreds) shows and generated no .json file
<details><summary>Log</summary>

```
2021-07-17 23:54:28,624 - tv_time_export.tv_time_extractor - INFO - Reading D:\Username\Downloads\tv_time_export-master\config.yaml
2021-07-17 23:54:28,625 - tv_time_export.request_handler - INFO - Logging in to Tv Time with user "Username"
2021-07-17 23:54:30,131 - tv_time_export.request_handler - INFO - Collecting data from Tv Time
2021-07-17 23:54:30,132 - tv_time_export.request_handler - INFO - Collecting all show ids
2021-07-17 23:54:31,681 - tv_time_export.request_handler - INFO - Collected 2 show ids
2021-07-17 23:54:33,160 - tv_time_export.request_handler - INFO - 002-002 - Collection state for "Marvel's Agents of S.H.I.E.L.D."
2021-07-17 23:54:33,253 - tv_time_export.request_handler - INFO - 001-002 - Done collecting state for "Marvel's Agents of S.H.I.E.L.D."
2021-07-17 23:54:33,306 - tv_time_export.request_handler - INFO - 001-002 - Collection state for "Loki"
2021-07-17 23:54:33,312 - tv_time_export.request_handler - INFO - 000-002 - Done collecting state for "Loki"
2021-07-17 23:54:33,313 - tv_time_export.request_handler - INFO - Logging out of Tv Time
2021-07-17 23:54:34,374 - tv_time_export.tv_time_extractor - INFO - Saving data to "D:\Username\Downloads\tv_time_export-master\Username_2021-07-17_23:54:34.json"
2021-07-17 23:54:34,374 - root - ERROR - [Errno 22] Invalid argument: 'D:\\Username\\Downloads\\tv_time_export-master\\Username_2021-07-17_23:54:34.json'
Traceback (most recent call last):
  File "C:\Users\Username\AppData\Local\Programs\Python\Python38\lib\site-packages\tv_time_export-1.0.10-py3.8.egg\tv_time_export\__main__.py", line 16, in main
    extractor.save_tv_show_states(tv_show_states)
  File "C:\Users\Username\AppData\Local\Programs\Python\Python38\lib\site-packages\tv_time_export-1.0.10-py3.8.egg\tv_time_export\tv_time_extractor.py", line 41, in save_tv_show_states
    with open(file_path, 'w+', errors='ignore') as file:
OSError: [Errno 22] Invalid argument: 'D:\\Username\\Downloads\\tv_time_export-master\\Username_2021-07-17_23:54:34.json'
```
</details>

**Comment by rhoriguchi** [comment]
So the code block which is the issue is:

https://github.com/rhoriguchi/tv_time_export/blob/62f263153eaf8d41475cd5a29dcda8ee44dfd4bf/tv_time_export/request_handler.py#L150-L167

Could you @DanGM96 add the HTML dome of this page <https://www.tvtime.com/user/USER_ID/profile> to the issue so i can debug it:

![image](https://user-images.githubusercontent.com/6047658/126066534-8106f01c-a09b-4d77-8484-ffce0b2d73a6.png)

**Comment by DanGM96** [comment]
Yeah, it might be easier to access [my profile](https://www.tvtime.com/en/user/7949169/profile) directly.

**Comment by rhoriguchi** [comment]
So I found the issue why you only get 2 shows. I didn't define any favorites so I defined a selector which didn't take that into account, so it just exported your "Recently watched" shows.

---

I currently don't have Windows to test but I'll switch from `os.path` to `pathlib` which should solve that issue. For the moment just manually specify a directory where to save the export and use the commit b5546f8cfd199f1e75439dafaaafb7228ba5fd18.

`config.yaml`
```
username: ****
password: ****
sava_path: /tmp/exports
```

**Comment by DanGM96** [comment]
This did fix the collection, but not the .json creation in Windows. But I have [WSL](https://docs.microsoft.com/en-us/windows/wsl/about) installed so it worked perfectly. Thanks for the help!

**Comment by rhoriguchi** [comment]
Could you please try ef82d2acfd83f0ee22bca78d5374db5fd5e2104d and attach the logs if it fails.

**Comment by DanGM96** [comment]
On Windows, right?
<details><summary>Log</summary>

```
[...]
2021-07-18 13:13:56,965 - tv_time_export.request_handler - INFO - Logging out of Tv Time
2021-07-18 13:13:58,200 - tv_time_export.tv_time_extractor - INFO - Saving data to "D:\Username\Downloads\tv_time_export-develop\Username_2021-07-18_13:13:58.json"
2021-07-18 13:13:58,200 - root - ERROR - [Errno 22] Invalid argument: 'D:\\Username\\Downloads\\tv_time_export-develop\\Username_2021-07-18_13:13:58.json'
Traceback (most recent call last):
  File "C:\Users\Username\AppData\Local\Programs\Python\Python38\lib\site-packages\tv_time_export-1.0.11-py3.8.egg\tv_time_export\__main__.py", line 16, in main
    extractor.save_tv_show_states(tv_show_states)
  File "C:\Users\Username\AppData\Local\Programs\Python\Python38\lib\site-packages\tv_time_export-1.0.11-py3.8.egg\tv_time_export\tv_time_extractor.py", line 42, in save_tv_show_states
    with open(file_path, 'w+') as file:
OSError: [Errno 22] Invalid argument: 'D:\\Username\\Downloads\\tv_time_export-develop\\Username_2021-07-18_13:13:58.json'
```
</details>

**Comment by rhoriguchi** [comment]
@DanGM96 I think I found the issue,  finally had a chance to test on Windows. Windows doesn't support `:` in file names. Should be fixed with commit a36544b1ee3506b5ab19b86fb213d4399d1893ce.

**Comment by DanGM96** [comment]
Nice find!
Past WSL output was `Username_2021-07-18_122838.json`
![image](https://user-images.githubusercontent.com/36743739/126562444-77db543c-97a2-4198-b77a-184e672691a4.png) Interesting way of resolving this `:` conflict.

Current Windows output:
![image](https://user-images.githubusercontent.com/36743739/126562837-236cfb92-59f1-485b-b687-098df7529a0b.png) No need to replace `-` but that's ok.

What I see right now is that the config.yaml `sava_path: /tmp/exports` is ignored, as it is just creating the .json in the root folder (it was like this before, I just didn't notice).
